### PR TITLE
Fix TaskIT

### LINKED ISSUE #64056: [CI] TasksIT testGetValidTask failure
**Build scan**: https://gradle-enterprise.elastic.co/s/m4ri3qe5dslwk

**Repro line**: `./gradlew ':client:rest-high-level:asyncIntegTestRunner' --tests "org.elasticsearch.client.TasksIT.testGetValidTask" -Dtests.seed=811D7AD2556E9BC7 -Dtests.security.manager=true -Dtests.locale=zh-CN -Dtests.timezone=Jamaica -Druntime.java=8` 

**Reproduces locally?**: No

**Applicable branches**: 7.9

**Failure history**: [it failed in the past but the errors were different then and maybe not all of them strictly related to this method](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-90d,mode:quick,to:now))&_a=(columns:!(_source),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'class:%22org.elasticsearch.client.TasksIT%22%20AND%20test:%22testGetValidTask%22'),sort:!(time,desc)))

**Failure excerpt**:
```
org.elasticsearch.client.TasksIT > testGetValidTask FAILED
    org.elasticsearch.client.ResponseException: method [GET], host [http://127.0.0.1:42117], URI [/_tasks/B7ZED5VwT_i5O_wdl3eN9A:57079], status line [HTTP/1.1 404 Not Found]
    {"error":{"root_cause":[{"type":"resource_not_found_exception","reason":"task [B7ZED5VwT_i5O_wdl3eN9A:57079] isn't running and hasn't stored its results"}],"type":"resource_not_found_exception","reason":"task [B7ZED5VwT_i5O_wdl3eN9A:57079] isn't running and hasn't stored its results"},"status":404}
        at __randomizedtesting.SeedInfo.seed([811D7AD2556E9BC7:8155551EC6C3380C]:0)
        at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:302)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:272)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:246)
        at org.elasticsearch.client.ReindexIT.lambda$checkCompletionStatus$0(ReindexIT.java:504)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:939)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:924)
        at org.elasticsearch.client.TasksIT.testGetValidTask(TasksIT.java:124)
```

### PR DESCRIPTION
**Pull request by tlrx**
This took me a bit of time so I'll try to describe what happened. The `TaskIT.testGetValidTask()` test failed twice this year ([1](https://gradle-enterprise.elastic.co/s/m4ri3qe5dslwk), [2](https://gradle-enterprise.elastic.co/s/bzjjapdkctkes)) with a "resource not exception" indicating that a task was not correctly created (the task is a reindexing task).

The elasticsearch logs show the following warning messages for both builds failures:
> WARN ][o.e.t.LoggingTaskListener] [asyncIntegTest-0] 56987 failed with exception
>  org.elasticsearch.index.mapper.MapperException: the [enabled] parameter can't be updated for the object mapping [response]

It indicates that the `.tasks` index was already created with a mapping for the `response` object. Since indices are deleted between tests there was no reason the index already exist when the test is executed. 

More over the index is a system index and should be appear as `[.tasks] creating index, cause [auto(task api)]` in logs when it is created but in the failures the index creation is logged with cause `[auto(bulk api)]`, indicating that some indexing operation somewhere (re)created the tasks index.

The only other action that could create a document in the .tasks index is a `DeleteJobAction` with a `wait_for_completion` set to `false` as this action overrides `ActionRequest#getShouldStoreResult()` which allows to store a task result for the action.

Such an action is triggered by the test `MachineLearningIT.testDeleteJob_GivenWaitForCompletionIsFalse()` which does not wait for the task result to be indexed. The REST API call it executes also [does not wait at all](https://github.com/elastic/elasticsearch/blob/efecaed611c006527e6959bf6c731091f5d12783/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/job/RestDeleteJobAction.java#L54) for any result when `wait_for_completion=false`. 

The test failure in #64056 can be reproduce when the test in `MachineLearningIT` is executed soon before the test `TaskId`.

This pull request makes the test in `MachineLearningIT` wait for the task to complete.

Closes #64056

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java [hunk 1]
```diff
@@ -302,7 +306,18 @@ public void testDeleteJob_GivenWaitForCompletionIsFalse() throws Exception {
...
         assertNull(response.getAcknowledged());
-        assertNotNull(response.getTask());
+
+        final TaskId taskId = response.getTask();
+        assertNotNull(taskId);
+
+        // When wait_for_completion=false the DeleteJobAction stored the task result in the .tasks index. In tests we need to wait
+        // for the delete job task to complete, otherwise the .tasks index could be created during the execution of a following test.
+        final GetTaskRequest taskRequest = new GetTaskRequest(taskId.getNodeId(), taskId.getId());
+        assertBusy(() -> {
```
Comment (original-brownbear) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java:316]: Should this get more than a 10s wait?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java [hunk 2]
```diff
@@ -302,7 +306,18 @@ public void testDeleteJob_GivenWaitForCompletionIsFalse() throws Exception {
         DeleteJobResponse response = execute(deleteJobRequest, machineLearningClient::deleteJob, machineLearningClient::deleteJobAsync);
 
         assertNull(response.getAcknowledged());
-        assertNotNull(response.getTask());
+
+        final TaskId taskId = response.getTask();
```
Comment (original-brownbear) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java:310]: One thing I don't fully understand is, how is this not caught by us ensure that no tasks are running after a test? Shouldn't this task show up in `_cat/tasks` and the wait we do here run there as a result?

**Review by original-brownbear — COMMENTED** [review]
I have one question :)

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java (hunk 2 above)
Comment (tlrx) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java:310]: The way the test works (before this PR) basically fire a job deletion and forgets about it. The delete job action is quite complicated and involves multiple steps, as well as storing the task result, so I suspect that at some point the task manager has no in-flight active tasks and allows to move to the next tests but the store result indexing task is not yet triggered (an registered) and is executed much later.

Comment (tlrx) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java:310]: (the test failure can be reproduced by executing the two tests one after the other)

Comment (original-brownbear) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java:310]: > The way the test works (before this PR) basically fire a job deletion and forgets about it. The delete job action is quite complicated and involves multiple steps, as well as storing the task result, so I suspect that at some point the task manager has no in-flight active tasks and allows to move to the next tests but the store result indexing task is not yet triggered (an registered) and is executed much later.

Got it, that makes sense, thanks! I wonder if the temporary absence of any task should be interpreted as a bug ... regardless LGTM :)

**Review by original-brownbear — APPROVED** [review]

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java (hunk 1 above)
Comment (tlrx) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java:316]: Forgot this one, sorry. Yes, 30L sounds better suited.

**Comment by tlrx** [comment]
Thanks Armin!

**Pull request merged**